test/BarcodeTest.php

```diff
@@ -18,6 +18,22 @@
  */
 class BarcodeTest extends \PHPUnit_Framework_TestCase
 {
+    public function provideBarcodeConstructor()
+    {
+        return [
+            [null],
+            [[]],
+        ];
+    }
+    /**
+     * @dataProvider provideBarcodeConstructor
+     */
+    public function testBarcodeConstructorIsNullOrArrayButNoOptionsKey($options)
```

Comment:
Don't detail constructor types in the function name, this is already provided by the provider.